app/plugin_key_value_store.go

```diff
@@ -23,18 +23,20 @@ func (a *App) SetPluginKey(pluginId string, key string, value []byte) *model.App
 }
 
 func (a *App) SetPluginKeyWithExpiry(pluginId string, key string, value []byte, expireInSeconds int64) *model.AppError {
-
 	if expireInSeconds > 0 {
 		expireInSeconds = model.GetMillis() + (expireInSeconds * 1000)
 	}
 
 	kv := &model.PluginKeyValue{
 		PluginId: pluginId,
-		Key:      getKeyHash(key),
+		Key:      key,
 		Value:    value,
 		ExpireAt: expireInSeconds,
 	}
 
+	// First try deleting hashed key, then set using unhashed key
```

Comment:
I'm inclined to think we should insert the new key and then try to delete the old one (logging but otherwise ignoring any error). This avoids data loss in the event of an extreme edge case.